examples/fluids/qfunctions/channel.h

```diff
@@ -83,22 +83,21 @@ CEED_QFUNCTION(ICsChannel)(void *ctx, CeedInt Q,
   for (CeedInt i=0; i<Q; i++) {
     const CeedScalar x[] = {X[0][i], X[1][i], X[2][i]};
     State s = Exact_Channel(3, 0., x, 5, ctx);
-    if (context->newtonian_ctx.is_primitive) {
-      q0[0][i] = s.Y.pressure;
-      for (CeedInt j=0; j<3; j++)
-        q0[j+1][i] = s.Y.velocity[j];
-      q0[4][i] = s.Y.temperature;
-    } else {
-      q0[0][i] = s.U.density;
-      for (CeedInt j=0; j<3; j++)
-        q0[j+1][i] = s.U.momentum[j];
-      q0[4][i] = s.U.E_total;
-    }
+    CeedScalar q[5] = {0};
+    if (context->newtonian_ctx.is_primitive)
+      UnpackState_Y(s.Y, q);
+    else
+      UnpackState_U(s.U, q);
```

Comment:
I like the `UnpackState*` functions! I think we could move this to a function pointer method similar to what I did with the `newtonian.h` boundary integral QFs, where we set a generic `SetStateFromQi` function at the start of the routine and just use that throughout, `qi` representing whatever state variables the solver is using at the time.

Reply:
I'd want to do a performance test before using function pointers in perf-critical code (like the main volume integrals). I'm less concerned elsewhere. It's possible to transform so the key function pointers are passed as arguments to a function that is meant to be inlined (topic of another PR if we want to go that route).